cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,14 +60,39 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
-                 /usr
-                 /usr/local
-                 /opt
-                 $ENV{HOME}
-                 $ENV{HOME}/.local
-                 $ENV{HOME}/.local/opt
-                 $ENV{HOME}/bin)
+    # Paths that are used to find installed Zephyr SDK versions
+    SET(ZEPHYR_SDK_SEARCH_PATHS
+        /usr
+        /usr/local
+        /opt
+        $ENV{HOME}
+        $ENV{HOME}/.local
+        $ENV{HOME}/.local/opt
+        $ENV{HOME}/bin)
+
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS ${ZEPHYR_SDK_SEARCH_PATHS})
+
+    # Remove duplicate entries and sort naturally in descending order.
+    list(REMOVE_DUPLICATES Zephyr-sdk_CONSIDERED_VERSIONS)
+    list(SORT Zephyr-sdk_CONSIDERED_VERSIONS COMPARE NATURAL ORDER DESCENDING)
+
+    # Loop over each found Zepher SDK version until one is found that is compatible.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_VERSIONS})
+      find_package(Zephyr-sdk ${ZEPHYR_SDK_CANDIDATE} EXACT QUIET CONFIG PATHS ${ZEPHYR_SDK_SEARCH_PATHS})
+
+      if ("${Zephyr-sdk_FOUND}" EQUAL "1")
+        break()
```

Comment:
Can't do a return because this is not a function. And in actuality if you do a return here, it then breaks the whole build because it no longer caches the zephyr SDK directory:
```
Kconfig header saved to '/tmp/aa/zephyr/samples/hello_world/_AA/zephyr/include/generated/autoconf.h'
-- Configuring done
You have changed variables that require your cache to be deleted.
Configure will be re-run and you may have to reset some variables.
The following variables have changed:
CMAKE_C_COMPILER= /opt/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMAKE_C_COMPILER= /opt/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMAKE_C_COMPILER= /opt/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc

Loading Zephyr default modules (Zephyr repository).
-- Application: /tmp/aa/zephyr/samples/hello_world
-- CMake version: 3.25.2
-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.9") found components: Interpreter 
-- Cache files will be written to: /home/jamie/.cache/zephyr
-- Zephyr version: 3.2.99 (/tmp/aa/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
CMake Error at /tmp/aa/zephyr/cmake/modules/extensions.cmake:2520 (message):
  BOARD is not being defined on the CMake command-line, in the environment or
  by the app.
Call Stack (most recent call first):
  /tmp/aa/zephyr/cmake/modules/boards.cmake:52 (zephyr_check_cache)
  /tmp/aa/zephyr/cmake/modules/zephyr_default.cmake:113 (include)
  /tmp/aa/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /tmp/aa/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:117 (include_boilerplate)
  /home/jamie/sysbuild/zephyr/share/zephyr-package/cmake/zephyr_package_search.cmake:117 (include)
  /home/jamie/sysbuild/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:152 (check_zephyr_package)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FAILED: build.ninja 
/usr/bin/cmake --regenerate-during-build -S/tmp/aa/zephyr/samples/hello_world -B/tmp/aa/zephyr/samples/hello_world/_AA
ninja: error: rebuilding 'build.ninja': subcommand failed
｢22 /tmp/aa/zephyr/samples/hello_world/_AA (sdkverfix)｣$ ninja
ninja: error: build.ninja:30: loading 'CMakeFiles/rules.ninja': No such file or directory
include CMakeFiles/rules.ninja
                              ^ near here

```